darshan-util/pydarshan/darshan/datatypes/heatmap.py

```diff
@@ -113,9 +117,15 @@ def to_df(self, op: str, interval_index: bool = True):
             columns = pd.IntervalIndex.from_breaks(breaks)
         else:
             columns = np.arange(nbins)
-            
-        df = pd.DataFrame.from_dict(data, orient='index', columns=columns)
-        df.index.name = "rank"
 
-        return df
+        df_list = []
+        for op in ops:
+            data = self._data[op]
+            df = pd.DataFrame.from_dict(data, orient='index', columns=columns)
+            df.index.name = "rank"
+            df_list.append(df)
+
+        # sum the dataframes
+        hmap_df = functools.reduce(lambda x, y: x.add(y), df_list)
+        return hmap_df
```

Comment:
This `functools` business seems like a rather complicated way to write `sum(df_list)`. This diff passes the tests and is  more readable I'd say:

```diff
index 580cf17d..79cfc3af 100644
--- a/darshan-util/pydarshan/darshan/datatypes/heatmap.py
+++ b/darshan-util/pydarshan/darshan/datatypes/heatmap.py
@@ -1,4 +1,3 @@
-import functools
 from typing import Sequence
 
 import numpy as np
@@ -126,6 +125,5 @@ class Heatmap():
             df_list.append(df)
 
         # sum the dataframes
-        hmap_df = functools.reduce(lambda x, y: x.add(y), df_list)
-        return hmap_df
+        return sum(df_list)

```